Source/WebCore/page/TextIndicator.cpp

```diff
@@ -69,30 +70,44 @@ Ref<TextIndicator> TextIndicator::create(const TextIndicatorData& data)
 
 RefPtr<TextIndicator> TextIndicator::createWithRange(const SimpleRange& range, OptionSet<TextIndicatorOption> options, TextIndicatorPresentationTransition presentationTransition, FloatSize margin)
 {
-    RefPtr frame = range.startContainer().document().frame();
+    auto rangeToUse = range;
+    if (options.contains(TextIndicatorOption::UseUserSelectAllCommonAncestor)) {
+        if (auto rangeNode = commonInclusiveAncestor<ComposedTree>(range); rangeNode) {
```

Comment:
This can just be `if (auto* commonAncestor = commonInclusiveAncestor<ComposedTree>(range))`.

(I would also rename `rangeNode` to `commonAncestor`)